### PR TITLE
[mp] Contributing IO Class / Google Sheets Docs

### PR DESCRIPTION
**Pull request by mattppal**
# Description

This PR:
- Adds documentation for the Google Sheets IO Class.
- Adds a high-level overview for the process of contributing an IO class.
- Adds documentation for contributing to docs (very meta haha)
- Makes small changes to other contribution docs.

# How Has This Been Tested?

- [x] `mintlify dev`
- [x] `mintlify broken-links`

# Checklist
- [x] The PR is tagged with proper labels (bug, enhancement, feature, documentation)
- [x] I have performed a self-review of my own code
- [ ] I have added unit tests that prove my fix is effective or that my feature works
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
  - [x] If new documentation has been added, relative paths have been added to the appropriate section of `docs/mint.json`

### CODE REVIEW

File: docs/contributing/documentation/overview.mdx [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+<Frame>
+  <img src="https://github.com/mage-ai/assets/blob/main/contributing-docs.png?raw=True" />
+</Frame>
+
+## Level 2: Add complex components 💪
+
+### Images
+
+To add images to the Mage docs, you'll need to contact a team member or host the images in GitHub. Our assets currently live in a private repo to keep the docs repo lightweight.
```
Comment (dy46) [inline docs/contributing/documentation/overview.mdx:31]: the assets repo is actually public, but not sure what the process is for contributing to it

File: mage_ai/data_preparation/templates/data_exporters/google_sheets.py [hunk 2]
```diff
@@ -33,4 +33,4 @@ def export_to_google_sheet(df: DataFrame, **kwargs) -> None:
     GoogleSheets.with_config(ConfigFileLoader(config_path, config_profile)).export(
         df, 
         sheet_url=sheet_url
-    )
+    )
```
Comment (dy46) [inline mage_ai/data_preparation/templates/data_exporters/google_sheets.py:36]: did you mean to make this change? we usually keep an empty line at the end of a file

File: docs/_snippets/contribution-process.mdx [hunk 3]
```diff
@@ -0,0 +1,18 @@
+<Steps>
+    <Step title="Fork the Mage GitHub repo">
+    Visit [GitHub](https://github.com/mage-ai/mage-ai) and fork the Mage repo to your machine.
```
Comment (dy46) [inline docs/_snippets/contribution-process.mdx:3]: might just be semantics, but you fork the repo and then clone it to your machine

File: docs/_snippets/contribution-process.mdx [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    </Step>
+    <Step title="Create a branch">
+    Create a branch in your forked repo and commit your changes.
+    </Step>
+    <Step title="Create a pull request">
+    Once you've [made and tested your changes](/contributing/development-environment), create a [pull request](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request-from-a-fork) on the Mage repo.
+    </Step>
+    <Step title="Tag Mage team members for a review">
+    Add members for review in GitHub, please tag one of the following:
```
Comment (dy46) [inline docs/_snippets/contribution-process.mdx:12]: maybe add yourself for docs

File: docs/contributing/documentation/overview.mdx (hunk 1 above)
Comment (mattppal) [reply docs/contributing/documentation/overview.mdx:31]: Hm yeah maybe we keep this process for now and cross that bridge when we get there.

**Pull request merged**